conans/test/integration/cmake_targets_test.py

```diff
@@ -49,6 +49,37 @@ def package(self):
 
 @attr("slow")
 class CMakeTargetsTest(unittest.TestCase):
+    def transitive_flags_test(self):
+        client = TestClient()
+        conanfile = """from conans import ConanFile
+class Charlie(ConanFile):
+    def package_info(self):
+        self.cpp_info.sharedlinkflags = ["CharlieFlag"]
+"""
+        client.save({"conanfile.py": conanfile})
+        client.run("create . Charlie/0.1@user/testing")
+        conanfile = """from conans import ConanFile
+class Beta(ConanFile):
+    requires = "Charlie/0.1@user/testing"
+    def package_info(self):
+        self.cpp_info.sharedlinkflags = ["BetaFlag"]
+"""
+        client.save({"conanfile.py": conanfile})
+        client.run("create . Beta/0.1@user/testing")
+        conanfile = """from conans import ConanFile, load
+import os
+class Alpha(ConanFile):
+    requires = "Beta/0.1@user/testing"
+    generators = "cmake"
+    def build(self):
+        cmake = load(os.path.join(self.build_folder, "conanbuildinfo.cmake"))
+        self.output.info()
```

Comment:
This `build` function is not needed (at least the test passes without it), I'm not sure if having this function introduces a perceptible overhead in performance.